source/configure/calls-offloader-setup.md

```diff
@@ -0,0 +1,929 @@
+   ```
+
+## Configuration
+
+### API Configuration
+
+The API section controls how the service accepts requests:
+
+- **http.listen_address**: The address and port where the service listens (default: `:4545`)
+- **http.tls.enable**: Whether to use TLS encryption for the API
+- **security.allow_self_registration**: Allow clients to self-register for job management
+- **security.enable_admin**: Enable admin functionality
+- **security.admin_secret_key**: Secret key for admin authentication (change from default!)
+
+### Store Configuration
+
+Controls persistent data storage:
+
+- **data_source**: Path to directory for storing job metadata and state
+
+### Jobs Configuration
+
+Controls job processing behavior:
+
+- **api_type**: Job execution backend (`docker` or `kubernetes`)
+- **max_concurrent_jobs**: Maximum number of simultaneous recording/transcription jobs
+- **failed_jobs_retention_time**: How long to keep failed job data before cleanup
+- **image_registry**: Docker registry for job runner images (typically `mattermost`)
+
+### Logger Configuration
+
+Controls logging output:
+
+- **enable_console**: Log to console output
+- **console_json**: Use JSON format for console logs
+- **console_level**: Log level for console (DEBUG, INFO, WARN, ERROR)
+- **enable_file**: Log to file
+- **file_location**: Path to log file
+- **enable_color**: Use colored output for console logs
+
+### Private Network Configuration
+
+When the Mattermost deployment is running in a private network, additional configuration may be necessary for the jobs spawned by the calls-offloader service to reach the Mattermost server.
+
+In such cases, you can override the site URL used by recorder jobs or transcriber jobs to connect to Mattermost by setting the following environment variables on the Mattermost server:
+
+- **MM_CALLS_RECORDER_SITE_URL**: Override the site URL used by recording jobs
+- **MM_CALLS_TRANSCRIBER_SITE_URL**: Override the site URL used by transcription jobs
+
+Example configuration:
+
+Create or edit the Mattermost environment file (`/opt/mattermost/config/mattermost.environment`):
+
+```bash
+MM_CALLS_RECORDER_SITE_URL="http://internal-mattermost-server:8065"
+MM_CALLS_TRANSCRIBER_SITE_URL="http://internal-mattermost-server:8065"
+```
+
+Then ensure your Mattermost systemd service references this environment file:
+
+```ini
+[Unit]
+Description=Mattermost
+After=network.target
+
+[Service]
+Type=notify
+EnvironmentFile=/opt/mattermost/config/mattermost.environment
+ExecStart=/opt/mattermost/bin/mattermost
+TimeoutStartSec=3600
+KillMode=mixed
+Restart=always
+RestartSec=10
+WorkingDirectory=/opt/mattermost
+User=mattermost
+Group=mattermost
+
+[Install]
+WantedBy=multi-user.target
+```
+
+This is particularly useful when:
+
+- The calls-offloader service runs in a different network segment than clients
+- Internal DNS resolution differs from external URLs
+- You need to use internal load balancer endpoints for job communication
```

Comment:
It's also quite useful if the Mattermost SiteURL is HTTPs, since it avoids the need to deal with client-side certificates, essentially bypassing all of that complexity.